// @wordpress/components/build/custom-select-control-v2/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CustomSelect = CustomSelect;
exports.CustomSelectContext = void 0;
exports.CustomSelectItem = CustomSelectItem;
var _react = require("react");
var Ariakit = _interopRequireWildcard(require("@ariakit/react"));
var _element = require("@wordpress/element");
var _i18n = require("@wordpress/i18n");
var Styled = _interopRequireWildcard(require("./styles"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/**
 * External dependencies
 */
// eslint-disable-next-line no-restricted-imports

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

const CustomSelectContext = exports.CustomSelectContext = (0, _element.createContext)(undefined);
function defaultRenderSelectedValue(value) {
  const isValueEmpty = Array.isArray(value) ? value.length === 0 : value === undefined || value === null;
  if (isValueEmpty) {
    return (0, _i18n.__)('Select an item');
  }
  if (Array.isArray(value)) {
    return value.length === 1 ? value[0] :
    // translators: %s: number of items selected (it will always be 2 or more items)
    (0, _i18n.sprintf)((0, _i18n.__)('%s items selected'), value.length);
  }
  return value;
}
function CustomSelect({
  children,
  defaultValue,
  label,
  onChange,
  size = 'default',
  value,
  renderSelectedValue,
  ...props
}) {
  const store = Ariakit.useSelectStore({
    setValue: nextValue => onChange?.(nextValue),
    defaultValue,
    value,
    // fix for Safari bug: https://github.com/WordPress/gutenberg/issues/55023#issuecomment-1834035917
    virtualFocus: false
  });
  const {
    value: currentValue
  } = store.useState();
  const computedRenderSelectedValue = renderSelectedValue !== null && renderSelectedValue !== void 0 ? renderSelectedValue : defaultRenderSelectedValue;
  return (0, _react.createElement)(_react.Fragment, null, (0, _react.createElement)(Styled.CustomSelectLabel, {
    store: store
  }, label), (0, _react.createElement)(Styled.CustomSelectButton, {
    ...props,
    size: size,
    hasCustomRenderProp: !!renderSelectedValue,
    store: store
  }, computedRenderSelectedValue(currentValue), (0, _react.createElement)(Ariakit.SelectArrow, null)), (0, _react.createElement)(Styled.CustomSelectPopover, {
    gutter: 12,
    store: store,
    sameWidth: true
  }, (0, _react.createElement)(CustomSelectContext.Provider, {
    value: {
      store
    }
  }, children)));
}
function CustomSelectItem({
  children,
  ...props
}) {
  const customSelectContext = (0, _element.useContext)(CustomSelectContext);
  return (0, _react.createElement)(Styled.CustomSelectItem, {
    store: customSelectContext?.store,
    ...props
  }, children !== null && children !== void 0 ? children : props.value, (0, _react.createElement)(Ariakit.SelectItemCheck, null));
}
